### ISSUE TITLE
Screen Tracking data slightly polluted when using a native navigation

### ISSUE THREAD

**Issue by GioLogist** [invertase/react-native-firebase#2994]
### Issue Description

I was originally going to post this over in `react-native-navigation`, but I realized its better suited here. As the issue seems to be related to automatically created screen events, being generated by Firebase Analytics.

Currently, screen tracking is not documented at all. Aside from that, when attempting to set it up, there are inconsistencies that disrupt the flow of data.

### Steps to Reproduce / Code Snippets / Screenshots

Below is a simple snippet, used to record screen events.

```js
Navigation.events().registerComponentDidAppearListener(({ componentId, componentName, passProps }) => {
  analytics().setCurrentScreen(componentName, componentName)
});
```

Nice, now it shows my screen (EventList)

![image](https://user-images.githubusercontent.com/4120754/70645636-36008880-1bfa-11ea-81bb-0cdfc09cadc2.png)

However, it in between navigating, it also recorded an event for `RNNComponentViewController` (native view).

![image](https://user-images.githubusercontent.com/4120754/70645726-56304780-1bfa-11ea-8b55-7a89aff6c67a.png)

Meaning, that, during my navigation to `EventList`, my "previous screen" was set to `RNNComponentViewController` and not the actual previous screen.

![image](https://user-images.githubusercontent.com/4120754/70645809-8677e600-1bfa-11ea-8aac-08980ee6762a.png)

This seems to be due to the native views automatically recording screen events, since  `react-native-navigation` uses a native navigation, unlike `react-navigation`.

---
### Environment
* React Native Firebase version: 6.2.0
* React Native version: 0.61.5
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator

**Comment by GioLogist** [comment]
## Additional thoughts

My initial inclination was to disable screen reporting. However, after reading a [couple](https://stackoverflow.com/questions/44746794/disabling-firebase-automatic-screen-reporting) of [different](https://segment.com/docs/connections/destinations/catalog/firebase/#screen) explanations and testing myself, it seems segment was correct when saying:

> Google Analytics for Firebase iOS does NOT support the case of manual-only screen reporting. Firebase only supports automatic + manual screen reporting or no screen reporting at all.

So, the only solution I can think of is to handle this directly, on the native side of things somehow.

**Comment by mikehardy** [comment]
What would you propose as a solution in this module? I'm having trouble thinking of a solution that isn't react-native-navigation related. I don't have this problem with react-navigation at least, and I think if you do the useScreens trick there you still get native-like performance?

**Comment by GioLogist** [comment]
I'm equally as stumped.

I can't see how `react-native-navigation` could avoid this? They're using native views, because of the way it's injected. Firebase is "technically" doing its job of automatically recording this. If I'm not mistaken, `react-navigation` (even with `react-native-screens`) injects itself into a single native UI View. So, technically it never navigates away, as far as the automatic tracking is concerned.

Perhaps there's a way to pass UI class names to Firebase Analytics, IE: `ignoreScreenClassNames` (array).

**Comment by GioLogist** [comment]
Not stale

**Comment by GazzaHazza** [comment]
Not stale

**Comment by Ehesp** [comment]
I don't know what we can do here I'm afraid. We can only use the methods provided by the native SDKs (which are exposed), so maybe worth creating an issue on the other libraries or raise a Firebase support request to see what they say?

**Comment by shamilovtim** [comment]
Landed in #6588
